GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1168,11 +1173,10 @@ private void UpdatePlaceholders()
                             // The delete will be attempted if one of the following is true
                             // 1. not in the projection anymore
                             // 2. in the projection but is not a folder in the projection
-                            // 3. Folder no longer exist on disk - this is to remove the tombstone
+                            // 3. Folder placeholder is a possible tombstone
                             if (!isProjected ||
                                 !isFolder ||
-                                (!GVFSPlatform.Instance.KernelDriver.EnumerationExpandsDirectories &&
-                                !this.context.FileSystem.DirectoryExists(Path.Combine(this.context.Enlistment.WorkingDirectoryRoot, folderPlaceholder.Path))))
+                                folderPlaceholder.IsPossibleTombstoneFolder)

```

Comment:
Before we checked if the file exists before attempting to delete it. If this is a `PossibleTombstone`, do we always delete the folder? Do we need to check if this "possible" tombstone is actually a tombstone?

Reply:
We just attempt the delete in `RemoveFolderPlaceholderIfEmpty` and it will get a `FSResult.FileOrPathNotFound` if it isn't a tombstone which saves a check by calling the directory exists or trying to read the file attributes and check for tombstones.